### ISSUE TITLE
Re-sort listview after some change

### ISSUE THREAD

**Issue by sith** [edvin/tornadofx#1016]
How can I re-sort elements of ListView after some event occurred( e.g. checkbox is selected)?

Is there a manual that describes how to apply filtering and sorting?

**Comment by edvin** [comment]
TornadoFX has a `SortedFilteredList` which should help you out :) Check the guide for usage examples.

**Comment by sith** [comment]
I figured it out but this gives no results https://edvin.gitbooks.io/tornadofx-guide/?q=SortedFilteredList.

Which guide are you referring too?

**Comment by sith** [comment]
sos!?

**Comment by edvin** [comment]
Looks like that info hasn't been moved to the guide yet, so it's in the Wiki still:

https://github.com/edvin/tornadofx/wiki/Utilities

**Comment by sith** [comment]
It is still not clear how to trigger re-sorting when some event happened

**Comment by edvin** [comment]
Not sure what you mean. The page I linked to states "You can force a refilter by calling the `refilter()` function.". Is that not what you're after?

**Comment by sith** [comment]
My listview has list of items which checkboxes. I want to show un-checked items on the top of the view and checked at the bottom. When use selects-unselects checkbox I want list to be re-sorted so item moves to a proper area. I am not sure refilter() will do it.

**Comment by sith** [comment]
sos

**Comment by edvin** [comment]
Seems like you just want to sort the table based on the boolean field the checkbox is bound to, is that it?

**Comment by sith** [comment]
Yes exactly. It is a simple task list where I want to sort items based on it's status(open/done). So every time the task is mark as done I want to re-sort list so it goes down

**Comment by sith** [comment]
bump

**Comment by edvin** [comment]
I've updated the `TodoList` demo app with a solution to this :)

**Comment by sith** [comment]
This example works with tableview, is there way to make it work with listview?

**Comment by sith** [comment]
Looking at the code I see that comparator in SortedFilteredList is not set:
I created pull request

**Comment by sith** [comment]
I think the pull request is not necessary.  When I create SortedFilteredList I can add my comparator like this 
`SortedFilteredList<MyItem>().apply { sortedItems.comparatorProperty().bind(itemComparatorProperty) }`
after this I can call refilter() function when checkbox is selected/unselected.

**Comment by sith** [comment]
The follow up question is if there is a way to animate addition/deletion/resorting/refiltering items. As of now UI changes too quickly. It would be better if  modifications to listview were more visible so user can see what is going on

**Comment by edvin** [comment]
No, I don't think ListView has any built in animation capabilities. It's pretty easy to set the comparator, but we could sure make it more convenient. I'll look at your PR and see if that's the way to go or not :)
